src/Bridge/Doctrine/Orm/SubresourceDataProvider.php

```diff
@@ -175,6 +176,17 @@ private function buildQuery(array $identifiers, array $context, QueryNameGenerat
                     $qb->select($alias)
                         ->from($identifierResourceClass, $alias);
                     break;
+                // ONE_TO_ONE relation can be Unidirectional or Bidirectional. For Bidirectional mappedBy attribute is used.
+                case ClassMetadataInfo::ONE_TO_ONE:
```

Comment:
If there is no `mappedBy` we should go through the `default` section right? Catching a `MappingException` doesn't look like a good option here.

Reply:
yes, right. Alternative solution could be to check with `isset` for `$previousAssociationProperty` and `mappedBy` index, like

```
if (isset($classMetadata->associationMappings[$previousAssociationProperty]) 
    && isset($classMetadata->associationMappings[$previousAssociationProperty]['mappedBy'])) {
     $mappedBy = $classMetadata->getAssociationMapping($previousAssociationProperty)['mappedBy'];
    $previousAlias = "$previousAlias.$mappedBy";

     $qb->select($alias)
           ->from($identifierResourceClass, $alias);
     break;
}
```
which does not look clear here I think

Reply:
I'd prefer an isset test indeed. 

For example (it's the default section of the switch): 

```
default:
    $association = $classMetadata->getAssociationMapping($previousAssociationProperty);
    if (isset($association['mappedBy'])) {
      $previousAlias = "$previousAlias.$mappedBy";
      $qb->select($alias)
          ->from($identifierResourceClass, $alias);
      break;
    }

    $qb->select("IDENTITY($alias.$previousAssociationProperty)")
        ->from($identifierResourceClass, $alias);

```

Reply:
Do you suggest to not create special `ONE_TO_ONE` case, but just put it into default?

Reply:
Mhhh with a `ONE_TO_ONE` case, you'd either repeat the code of the default section or use a `goto` instruction:

```
    case ONE_TO_ONE:
        $association = $classMetadata->getAssociationMapping($previousAssociationProperty);
        if (!isset($association['mappedBy'])) {
            goto def;
        }
        // ...
    default:
       def:
        $qb->select("IDENTITY($alias.$previousAssociationProperty)")
        ->from($identifierResourceClass, $alias);
```

I think that a `MANY_TO_ONE` would work the same way as the `ONE_TO_ONE` (these are both `to-one`) and it's why I suggested to put everything in the `default` but you made me doubt :p.

Reply:
with `MANY_TO_ONE` there is only an unidirectional relation, and `mappedBy` is not used at all, so it shouldn't be the case.

Reply:
@acabala is right here, I test ClassMetadataInfo::MANY_TO_ONE (2) so it goes directly to default case and it works properly (the join is correct). The best part is when I dump some variable, the huge difference with OneToOne is that `$classMetadata->getAssociationMapping($previousAssociationProperty)['mappedBy'] `return NULL. Like @acabala said, in MANY_TO_ONE relation, the mappedBy is never used and is set to NULL by Doctrine. So we need to introduce a case ONE_TO_ONE for sure now. I think @soyuka  is rigth with using a goto to avoid code duplication. I am working on a solution and writing some unit test !